src/Facility.CodeGen.CSharp/CSharpServiceInfo.cs

```diff
@@ -110,6 +132,15 @@ private CSharpServiceInfo(ServiceInfo serviceInfo, CSharpServiceInfoSettings? se
 							validationErrors.Add(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(csharpAttribute.Name, parameter));
 					}
 				}
+				else if (descendant is ServiceExternalDtoInfo or ServiceExternalEnumInfo)
+				{
+					var allowed = new HashSet<string> { "namespace", "type" };
```

Comment:
I would prefer the we follow the pattern used above (at least in part) and elsewhere and collect the data here, e.g. in a dictionary by DTO or enum name. This avoids repeating the allowed parameters here and where they are used. (I'm not sure why we handle both services and fields in the same `if` above. It does seem like it would be better split out like you do here.)

Reply:
I'll go back here and attempt a little refactor to better conform with the existing pattern.

Reply:
I updated this in da8e871c2adaa1d971ab3ab0900d8645b57c34c0, but let me know if there's better ways of approaching this.